### PR TITLE
Charon hardware testing

### PR DESCRIPTION
**Pull request by tganley**
Merge Josh's changes from Charon testing

### CODE REVIEW

File: equipment_handler/source/ads7128.c [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ * @param[in] channel from 0 to 7 (location of specific temp sensor channels TBD)
+ * @param[out] temperature value in deg C
+ * @return
+ *      1 if error, 0 if successful
+ */
+uint8_t readSingleTemp(uint8_t channel, int* temperature){//note int8 max = 127
+
+    uint8_t txdata[3] = {0b00001000, 0x11, (0x00 + channel)};
+
+    if(i2c_Send(ADS7128_PORT, ADS7128_ADDR, 3, txdata) != 0){//set channel ID for readout
```
Comment (Taylorr82) [inline equipment_handler/source/ads7128.c:100]: shouldn't txdata be a pointer?

File: equipment_handler/source/pcal9538a.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ *      The power channel that is to be reset.
+ * @return
+ *      1 if error, 0 if successful.
+ */
+uint8_t triggerPowerChannelReset(Power_Channel channel){
+
+    uint8_t data[2] = {0b1, 0b0};
+    data[1] = 0b1 << channel;
+
+    if(i2c_Send(PCAL9538A_PORT, PCAL9538A_ADDR, 2, data) != 0){
```
Comment (Taylorr82) [inline equipment_handler/source/pcal9538a.c:61]: shouldn't data be a pointer

File: equipment_handler/source/ads7128.c (hunk 1 above)
Comment (tganley) [reply equipment_handler/source/ads7128.c:100]: It is?

File: equipment_handler/source/pcal9538a.c (hunk 2 above)
Comment (tganley) [reply equipment_handler/source/pcal9538a.c:61]: Name of array is pointer to array, no?

File: hardware_interface/source/skytraq_binary.c [hunk 3]
```diff
@@ -378,13 +402,14 @@ GPS_RETURNSTATE skytraq_configure_power_mode(skytraq_power_mode mode, skytraq_up
...
-GPS_RETURNSTATE skytraq_get_gps_time(uint8_t *reply) {
+GPS_RETURNSTATE skytraq_get_gps_time(uint8_t *time) {
     uint16_t length = 2;
     uint8_t payload[2];
+    uint8_t reply[100];
 
     *(uint16_t *)&(payload[0]) = QUERY_GPS_TIME;
 
-    return skytraq_send_message_with_reply(payload, length, reply);
+    return skytraq_send_message_with_reply(payload, length, reply, 100);
```
Comment (Taylorr82) [inline hardware_interface/source/skytraq_binary.c]: I'm not sure what's happening here. Where does the value of time get updated?

File: hardware_interface/source/skytraq_binary.c [hunk 4]
```diff
@@ -171,9 +185,9 @@ GPS_RETURNSTATE skytraq_send_message_with_reply(uint8_t *payload, uint16_t size,
         return UNKNOWN_ERROR;
     }
     bool cs_success = skytraq_verify_checksum(sentence);
-
+    cs_success = true;
     if (cs_success) {
-        strcpy((char *)reply, (char *)sentence);
+        memcpy((char *)reply, (char *)sentence, reply_len);
```
Comment (Taylorr82) [inline hardware_interface/source/skytraq_binary.c:193]: Good to use a bounded function for sure

File: hardware_interface/source/skytraq_binary.c [hunk 5]
```diff
@@ -89,7 +91,12 @@ bool skytraq_binary_init() {
  * @return GPS_RETURNSTATE Error explaining why the failure occurred
  */
 GPS_RETURNSTATE skytraq_send_message(uint8_t *paylod, uint16_t size) {
+    if(xSemaphoreTake(uart_mutex, 1000 * portTICK_PERIOD_MS) != pdTRUE) {
```
Comment (Taylorr82) [inline hardware_interface/source/skytraq_binary.c]: timeout should be a macro

File: equipment_handler/source/pcal9538a.c (hunk 2 above)
Comment (Taylorr82) [reply equipment_handler/source/pcal9538a.c:61]: Yeah I guess. This part of pointers always trips me up

File: hardware_interface/source/skytraq_binary.c (hunk 4 above)
Comment (tganley) [reply hardware_interface/source/skytraq_binary.c:193]: Strcpy was often missing the full reply cuz there were 0s in the data

File: hardware_interface/source/skytraq_binary.c (hunk 3 above)
Comment (tganley) [reply hardware_interface/source/skytraq_binary.c]: It doesn't yet. I did not finish writing out what this should return, hence my question about this in slack

File: hardware_interface/source/skytraq_binary.c (hunk 5 above)
Comment (tganley) [reply hardware_interface/source/skytraq_binary.c]: Fixed

**Review by Taylorr82 — APPROVED** [review]

**Comment by Taylorr82** [comment]
Shouldn't we integrate the changes from https://github.com/AlbertaSat/ex2_gps_software/pull/9 before we merge this?

**Pull request merged**